pco/src/data_types/floats.rs

```diff
@@ -6,52 +6,99 @@ use crate::constants::Bitlen;
 use crate::data_types::{split_latents_classic, FloatLike, Latent, NumberLike};
 use crate::describers::LatentDescriber;
 use crate::errors::{PcoError, PcoResult};
+use crate::float_mult_utils::FloatMultConfig;
 use crate::{
-  describers, float_mult_utils, float_quant_utils, ChunkConfig, ChunkMeta, FloatMultSpec,
-  FloatQuantSpec, Mode,
+  describers, float_mult_utils, float_quant_utils, mode::Bid, sampling, ChunkConfig, ChunkMeta,
+  FloatMultSpec, FloatQuantSpec, Mode,
 };
 
-fn choose_mode_and_split_latents<F: FloatLike>(
+use super::ModeAndLatents;
+
+fn filter_sample<F: FloatLike>(num: &F) -> Option<F> {
+  // We can compress infinities, nans, and baby floats, but we can't learn
+  // the mode from them.
+  if num.is_finite_and_normal() {
+    let abs = num.abs();
+    if abs <= F::MAX_FOR_SAMPLING {
+      return Some(abs);
+    }
+  }
+  None
+}
+
+fn detect_mode_and_split_latents<F: FloatLike>(
```

Comment:
We should add a unit test like the one you outlined in https://github.com/mwlon/pcodec/issues/194#issuecomment-2153285188, where `FloatMult::Enabled` and `FloatQuant::Enabled` are competing against each other and it's a close call.